common/prometheus-server/templates/alerts/_prometheus.alerts.tpl

```diff
@@ -21,145 +21,284 @@ groups:
+    expr: |
+      increase(prometheus_target_scrapes_exceeded_body_size_limit_total{prometheus="{{ include "prometheus.name" . }}"}[5m]) > 0
+    for: 15m
+    labels:
+      service: {{ default "metrics" $root.Values.alerts.service }}
+      support_group: {{ default "observability" $root.Values.alerts.support_group }}
+      severity: warning
+      playbook: docs/support/playbook/prometheus/rule_evaluation.html
+      meta: Prometheus `{{`{{ $labels.prometheus }}`}}` has dropped some targets that exceeded body size limit.
     annotations:
-      description: 'Prometheus `{{`{{ $labels.prometheus }}`}}` rule evaluation is slow'
-      summary: Prometheus rule evaluation is slow
+      description: |
+        Prometheus `{{`{{ $labels.prometheus }}`}}` as failed
+        `{{`{{ printf "%.0f" $value }}`}}` scrapes in the last 5m because
+        some targets exceeded the configured body_size_limit.
+      summary: Prometheus has dropped some targets that exceeded body size limit.
+
+  - alert: PrometheusScrapeSampleLimitHit
+    expr: |
+      increase(prometheus_target_scrapes_exceeded_sample_limit_total{prometheus="{{ include "prometheus.name" . }}"}[5m]) > 0
+    for: 15m
+    labels:
+      service: {{ default "metrics" $root.Values.alerts.service }}
+      support_group: {{ default "observability" $root.Values.alerts.support_group }}
+      severity: warning
+      playbook: docs/support/playbook/prometheus/rule_evaluation.html
+      meta: Prometheus `{{`{{ $labels.prometheus }}`}}` has failed scrapes that have exceeded the configured sample limit.
+    annotations:
+      description: |
+        Prometheus `{{`{{ $labels.prometheus }}`}}` as failed
+        `{{`{{ printf "%.0f" $value }}`}}` scrapes in the last 5m because
+        some targets exceeded the configured sample_limit.
+      summary: Prometheus has failed scrapes that have exceeded the configured sample limit.
 
   - alert: PrometheusWALCorruption
-    expr: increase(prometheus_tsdb_wal_corruptions_total{prometheus="{{ include "prometheus.name" . }}"}[5m]) > 0
+    expr: |
+      increase(prometheus_tsdb_wal_corruptions_total{prometheus="{{ include "prometheus.name" . }}"}[5m]) > 0
     labels:
-      context: availability
       service: {{ default "metrics" $root.Values.alerts.service }}
       support_group: {{ default "observability" $root.Values.alerts.support_group }}
       severity: info
-      playbook: 'docs/support/playbook/prometheus/wal.html'
-      meta: 'Prometheus `{{`{{ $labels.prometheus }}`}}` has `{{`{{ $value }}`}}` WAL corruptions.'
+      playbook: docs/support/playbook/prometheus/wal.html
+      meta: Prometheus `{{`{{ $labels.prometheus }}`}}` has WAL corruptions.
     annotations:
-      description: 'Prometheus `{{`{{ $labels.prometheus }}`}}` has `{{`{{ $value }}`}}` WAL corruptions.'
-      summary: Prometheus has WAL corruptions
+      description: |
+        Prometheus `{{`{{ $labels.prometheus }}`}}` has `{{`{{ $value }}`}}` WAL corruptions.
```

Comment:
The description does not really fit the alert expression. If the corruption increases, `$value` will be at a certain value. Once increasing stops, it will be back to normal. However, the corrupt WAL blocks are staying in place.

Maybe we should split this in two. This would rather be a `PrometheusWALCorruptionIncreasing` alert. Maybe we should have one for total too, not taking the increase in consideration?

Reply:
Fair. To my mind, there should only be one WAL corruption alert without `increase`. 
Like `prometheus_tsdb_wal_corruptions_total > 0` to indicate corrupt WAL blocks.